lnpbp-0002.md

```diff
@@ -15,261 +15,384 @@ Created: 2019-10-27
-5. Constructs necessary scripts and generates `scriptPubkey` of the required type. If OP_RETURN `scriptPubkey` format is
-   used, it MUST be serialized according to the following rules:
-   - only a single `OP_RETURN` code MUST be present in the `scriptPubkey` and it MUST be the first byte of it;
-   - it must be followed by 32-byte push of the public key value `P` from the step 2 of the algorithm, serialized
-     according to from [15]; if the resulting public key containing the commitment is non-square, a new public key MUST
-     be picked and procedure from step 4 MUST BE repeated once more.
-6. Constructs and stores an **extra-transaction proof** (ETP), which structure depends on the generated `scriptPubkey`
-   type:
-   a) value of `S`, corresponding to:
-      * single original public key for P2PK, P2PKH, P2WPK, P2WPK-in-P2SH and OP_RETURN-type of outputs;
-      * the original intermediary public key for P2TR (V1 witness Taproot output);
-      * sum of all public keys required to spend P2SH, P2WSH or P2WSH-in-P2SH output;
+### Commitment procedure
+
+The **committing party**, having a message `msg`, must:
+1. Collect all unique public key instances (named **original public key set** 
+   `P`) related to a given transaction output, defined as:
+   - a single public key for P2PK, P2PKH, P2WPK, P2WPK-in-P2SH type of outputs;
+   - an arbitrary public key (even with an unknown corresponding private key),
+     if `OP_RETURN` type of `scriptPubkey` is used;
+   - an intermediate public key for the Taproot-based (SegWit v1) output;
+   - all public keys extracted using 
+     [algorithm](#deterministic-public-key-extraction-from-bitcoin-script) from 
+     either:
+     * `witnessScript` contained in the `witness` stack for native v0 P2WSH
+       and legacy P2WSH-in-P2SH SegWit outputs;
+     * `redeemScript` contained in top  `scriptSig` for non-SegWit P2SH
+     * or bare (custom) script-based `scriptPubkey`
+2. Select a single public key `Po` from the set `P` of the original public keys 
+   for the tweaking procedure. It is advised that the corresponding private key
+   being controlled by the committing party, which will simplify future spending
+   of the output.
+3. Run [LNPBP-1 commitment procedure](https://github.com/LNP-BP/LNPBPs/blob/master/lnpbp-0001.md#commitment-procedure)
+   for the message `msg`, selected public key `Po`, public key set `P` and a
+   protocol-specific `tag`, provided by the upstream protocol using this 
+   standard.
+4. Construct necessary scripts and generates `scriptPubkey` of the required 
+   type. If OP_RETURN `scriptPubkey` is used, it MUST be serialized according to
+   the following rules:
+   - only a single `OP_RETURN` code MUST be present in the `scriptPubkey` and it
+     MUST be the first byte of it;
+   - it must be followed by 32-byte push of the public key value `P` from the 
+     step 2 of the algorithm, serialized according to from [15];
```

Comment:
Possibly change 'step 2 of the algorithm, serialized according to from [15];' to 'step 2 of the algorithm, serialized according to the procedure from [15];'